### ISSUE TITLE
Is the `Azure` provider termed too broadly?

### ISSUE THREAD

**Issue by smudge202** [herecydev/Rush#10]
For one reason or another, there are multiple Service Bus' available on Azure. The full blown [Service Bus](https://azure.microsoft.com/en-gb/services/service-bus/), and the one that serves as an extension of the [storage API](https://azure.microsoft.com/en-gb/services/storage/queues/).

[Event Hubs](https://azure.microsoft.com/en-gb/services/event-hubs/) might too be useful, and any number of future technologies could make good candidates.

As such, is the `Azure` term being used too broadly in the `features/azure` branch?

**Comment by herecydev** [comment]
I think there are 3 candidates, Queues, Topics and Event hubs. My plan was to get the infrastructure in a place such that the majority of concepts (observables, serialization, error handling, logging) can be reused and it's just the client that will be swapped out depending on type of message. So I fully expect to implement more azure technologies, it's definitely going to happen.

**Comment by smudge202** [comment]
I think the "Queues" candidate you've specified is actually duplicated in Azure.

See these links:

 * [Azure Service Bus](https://azure.microsoft.com/en-gb/services/service-bus/) which has Queues and Topics
* [Azure Storage Queue](https://azure.microsoft.com/en-gb/services/storage/queues/) which also has queues.

These two are completely independent services, namespaces, and NuGet packages.

**Comment by herecydev** [comment]
Yer, I've seen these before. I'm not sure how many people use the storage queue, I always found the lack of deadlettering and push based messaging a nonstarter for the services I build.

Perhaps changing the `Azure` into `AzureServiceBus` will clarify things?

**Comment by smudge202** [comment]
The naming change would work for me.

You may find more people than you realise using the Storage variant, because the NuGet packages for it were best at keeping up with dnx. Not checked since netcore.

**Comment by herecydev** [comment]
Yep it support `NETStandard 1.6, the API is reasonably similar. Just need to implement the message pump for the observers and deal with some deadletter logic.
